astropy/stats/histogram.py

```diff
@@ -31,10 +31,10 @@
 def calculate_bin_edges(
     a: ArrayLike,
     bins: int
-    | list[int | float]
+    | list[float]
```

Comment:
`list` is a mutable collection, so it is invariant, not covariant (see [PEP 483 (The Theory of Type Hints)](https://peps.python.org/pep-0483/)). This means that although `int` is a subtype of `float`, `list[int]` is not a subtype of `list[float]`. Using a covariant type such as `Sequence[float]` would be better, but it might be even better to look up annotations from `np.histogram_bin_edges()`. The same goes for `histogram()` too.